libsolidity/parsing/Scanner.cpp

```diff
@@ -768,6 +779,15 @@ Token::Value Scanner::scanNumber(char _charSeen)
 			scanDecimalDigits();  // optional
 			if (m_char == '.')
 			{
+				if (m_source.get(1) == '_')
```

Comment:
Can this go out of bounds?

Reply:
Probably in special crafted source codes (that are then invalid anyways). I've been adding a guard now.